File: 0299-bulls-and-cows/0299-bulls-and-cows.java
class Solution {
    public String getHint(String secret, String guess) {
        int bull = 0, cow = 0;
        int[] countSecret = new int[10];
        int[] countGuess = new int[10];

        for (int i = 0; i < secret.length(); i++) {
            char s = secret.charAt(i);
            char g = guess.charAt(i);
            if (s == g) {
                bull++;
            } else {
                countSecret[s - '0']++;
                countGuess[g - '0']++;
            }
        }

        for (int i = 0; i < 10; i++) {
            cow += Math.min(countSecret[i], countGuess[i]);
        }

        return bull + "A" + cow + "B";
    }
}

File: 0299-bulls-and-cows/0299-bulls-and-cows.py
class Solution(object):
    def getHint(self, secret, guess):
        bulls = 0
        cows = 0
        count_s = [0] * 10
        count_g = [0] * 10
        for i in range(len(secret)):
            s = secret[i]
            g = guess[i]
            if s == g:
                bulls += 1
            else:
                count_s[int(s)] += 1
                count_g[int(g)] += 1
        for i in range(10):
            cows += min(count_s[i] , count_g[i])
        return str(bulls) + "A" + str(cows) + "B"


        